transforms/components/index.js

```diff
@@ -64,6 +64,17 @@ const getRequiredProperties = properties => (
 
 const formatRequiredProperties = requiredProperties => requiredProperties.map(getPropertyName);
 
+const addDictionaryAdditionalProperties = typedef => {
+  if (typedef.type?.applications?.length === 2 && typedef.type.applications[0].name === 'string') {
```

Comment:
Seems like this line is causing tests to break. 😢 

```bash
SyntaxError: Unexpected token '.'
```

Apparently the reason is that [the optional chaining (`?.`) operator is not supported on Node versions inferior to 14](https://stackoverflow.com/questions/59574047/how-to-use-optional-chaining-in-node-js-12), and the tests are currently expected to run in both Node 10 and 12.